Bottom: 022d0e4473e17ba75c2ba9fbf0d6234ac853c416
Top:    c9bffef4aed312a07adc1f19bc92101694dcc7f8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-26 00:10:40 +0100

draw.c: Split "Gathering" from real drawing routines (Element Pins and Pads)


---

diff --git a/src/draw.c b/src/draw.c
index d71982f..b9a812e 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -226,6 +226,22 @@ name_callback (const BoxType * b, void *cl)
   return 0;
 }
 
+static void
+draw_element_pins_and_pads (ElementType *element)
+{
+  PAD_LOOP (element);
+  {
+    if (doing_pinout || doing_assy || FRONT (pad) || PCB->InvisibleObjectsOn)
+      DrawPad (pad);
+  }
+  END_LOOP;
+  PIN_LOOP (element);
+  {
+    DrawPin (pin);
+  }
+  END_LOOP;
+}
+
 static int
 EMark_callback (const BoxType * b, void *cl)
 {
@@ -1562,6 +1578,8 @@ DrawElementPackage (ElementTypePtr Element)
 void
 DrawElementPinsAndPads (ElementTypePtr Element)
 {
+  assert (Gathering);
+
   PAD_LOOP (Element);
   {
     if (doing_pinout || doing_assy || FRONT (pad) || PCB->InvisibleObjectsOn)
@@ -1851,7 +1869,7 @@ draw_element (ElementTypePtr element)
 {
   draw_element_package (element);
   draw_element_name (element);
-  DrawElementPinsAndPads (element);
+  draw_element_pins_and_pads (element);
 }
 
 /* ---------------------------------------------------------------------------
